### PR TITLE
Activity details

### PR DESCRIPTION
**Pull request by mattgle**
## Summary

- Activity details
- Pass Profile to TS

## Screenshots

image

## Github Issue

https://github.com/Psychedelic/plug-mobile/issues/526

### CODE REVIEW

File: src/screens/tabs/Profile/index.tsx [hunk 1]
```diff
@@ -80,7 +98,7 @@ const Profile = () => {
             text={t('common.change')}
             buttonStyle={styles.buttonStyle}
             textStyle={styles.buttonTextStyle}
-            onPress={modalRef.current?.open}
+            onPress={() => accountsModalRef.current?.open()}
```
Comment (ignaciosantise) [inline src/screens/tabs/Profile/index.tsx:102]:
```suggestion
            onPress={accountsModalRef.current?.open}
```

File: src/screens/tabs/Profile/modals/ActivityDetail/index.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  value: string;
+  onPress?: (address: string) => void;
+  showSelf?: boolean;
+  hideRow?: boolean;
+}
+
+const formatAddress = (address: string) =>
+  validateICNSName(address) ? address : shortAddress(address);
+
+const isMySelf = (address: string, currentWallet: Wallet) =>
```
Comment (ignaciosantise) [inline src/screens/tabs/Profile/modals/ActivityDetail/index.tsx]: we could move formatAddress and isMySelf to utils/ids, what u think? and reuse it in other places

also, `isMySelf` -> `isOwnAddress`?

File: src/screens/tabs/Profile/modals/ActivityDetail/index.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+              </Text>
+            )}
+          </Text>
+        </Touchable>
+      </View>
+    ) : null;
+  };
+
+  return (
+    <Modal adjustToContentHeight modalRef={modalRef}>
```
Comment (ignaciosantise) [inline src/screens/tabs/Profile/modals/ActivityDetail/index.tsx]: use the header prop from the modal

File: src/screens/tabs/Profile/index.tsx (hunk 1 above)
Comment (mattgle) [reply src/screens/tabs/Profile/index.tsx:102]: It was like that i had to change it cuz of TS

File: src/screens/tabs/Profile/modals/ActivityDetail/index.tsx (hunk 2 above)
Comment (mattgle) [reply src/screens/tabs/Profile/modals/ActivityDetail/index.tsx]: Nice

File: src/screens/tabs/components/ActivityItem/index.tsx [hunk 4]
```diff
@@ -29,57 +35,62 @@ const ActivityItem = ({
   logo,
   canisterId,
   details,
+  style,
   canisterInfo,
-}: Transaction) => {
+  onPress,
+  onlyDate,
```
Comment (ignaciosantise) [inline src/screens/tabs/components/ActivityItem/index.tsx]: `hideAddress`?

File: src/screens/tabs/Profile/index.tsx [hunk 5]
```diff
@@ -35,35 +34,39 @@ import ActivityDetail from './modals/ActivityDetail';
 import styles from './styles';
 
 function Profile({ navigation }: ScreenProps<Routes.PROFILE>) {
-  const dispatch = useAppDispatch();
   const accountsModalRef = useRef<Modalize>(null);
   const activityDetailModalRef = useRef<Modalize>(null);
   const transactionListRef = useRef(null);
+  const [selectedTransaction, setSelectedTransaction] = useState<
+    Transaction | undefined
+  >(undefined);
```
Comment (ignaciosantise) [inline src/screens/tabs/Profile/index.tsx]: check if this works
```suggestion
  const [selectedTransaction, setSelectedTransaction] = useState<Transaction>();
```

Comment (mattgle) [reply src/screens/tabs/Profile/index.tsx]: Done!

**Review by ignaciosantise — APPROVED** [review]
🎉 cmoooon

**Pull request merged**